adr/adr-responses-api.md

```diff
@@ -0,0 +1,52 @@
+# ADR: Adopt Llama Stack Responses API for client-side Agent orchestration
+
+## Context
+
+Red Hat is developing an agentic capability for its AI platforms, including RHEL AI and Red Hat OpenShift AI (RHOAI). The core reasoning engine will leverage the Llama Stack. We are faced with a choice between two available APIs within Llama Stack for implementing agentic logic:
+
+1. **Llama Stack Agents API:** An older API that provides an explicit, stateful `Agent` construct on the server. The server manages the agent's state and conversation history within a `session`. While this simplifies state management for the client, this API is not aligned with the broader AI ecosystem's direction.
+2. **Llama Stack Responses API:** A newer, stateless API designed for compatibility with the OpenAI Responses API format. This API accepts a list of tools and a chat history and returns a model-generated response, which may include one or more tool calls. It does not provide a server-side `Agent` or `session` construct, requiring the client to manage the state of the interaction.
```

Comment:
I don't think you can consider Responses API to be stateless. It still retains your history as a state, so you can refer to it at any time later.